### ISSUE TITLE
Zed does not respect mouse cursor size 

### ISSUE THREAD

**Issue by Barugon** [zed-industries/zed#15788]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I have the mouse cursor on my desktop set to medium size (Settings/Accessibility/Seeing/Cursor Size) but it's the default size in Zed.

### Environment

Zed: v0.146.5 (Zed)
OS: Linux Wayland fedora 40
Memory: 31.2 GiB
Architecture: x86_64
GPU: AMD Radeon RX 7800 XT (RADV NAVI32) || radv || Mesa 24.1.5

### If applicable, add mockups / screenshots to help explain present your vision of the feature

Here's how the cursor looks on my desktop.
![cursor_de](https://github.com/user-attachments/assets/c0d09dac-d2d0-4275-8b3d-e61a6a848344)

Here's how the cursor looks in Zed.
![cursor_zed](https://github.com/user-attachments/assets/5af4f99e-a35a-4281-be50-e7c4b63dea28)

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by m4saurabh** [comment]
What desktop environment you are using, are you using a hyprland or a some other wayland compositor.

**Comment by apricotbucket28** [comment]
What's the output of the following commands?
- `gdbus call -e -d org.freedesktop.portal.Desktop -o /org/freedesktop/portal/desktop -m org.freedesktop.portal.Settings.Read '"org.gnome.desktop.interface"' '"cursor-theme"'`
- `gdbus call -e -d org.freedesktop.portal.Desktop -o /org/freedesktop/portal/desktop -m org.freedesktop.portal.Settings.Read '"org.gnome.desktop.interface"' '"cursor-size"'`

**Comment by Barugon** [comment]
> Which desktop environment are you using, are you using hyprland or some other wayland compositor.

> OS: Linux Wayland fedora 40

**Comment by Barugon** [comment]
> * `gdbus call -e -d org.freedesktop.portal.Desktop -o /org/freedesktop/portal/desktop -m org.freedesktop.portal.Settings.Read '"org.gnome.desktop.interface"' '"cursor-theme"'`

(<<'Mocu-Black-Right'>>,)

> * `gdbus call -e -d org.freedesktop.portal.Desktop -o /org/freedesktop/portal/desktop -m org.freedesktop.portal.Settings.Read '"org.gnome.desktop.interface"' '"cursor-size"'`

(<<32>>,)

> And could you share your logs? They may contain some important debugging information.

<details><summary>Zed.log</summary><pre>

2024-08-05T17:21:36.685430837-07:00 [INFO] ========== starting zed ==========
2024-08-05T17:21:36.707942895-07:00 [INFO] perform;
2024-08-05T17:21:36.708157366-07:00 [INFO] Opening main db
2024-08-05T17:21:36.708286348-07:00 [INFO] read_command;
2024-08-05T17:21:36.708520035-07:00 [INFO] read_command;
2024-08-05T17:21:36.708910436-07:00 [INFO] socket reader;
2024-08-05T17:21:36.70935607-07:00 [INFO] new;
2024-08-05T17:21:36.710032898-07:00 [INFO] keep_updated;
2024-08-05T17:21:36.710510121-07:00 [INFO] Using git binary path: None
2024-08-05T17:21:36.741035132-07:00 [INFO] extensions updated. loading 7, reloading 0, unloading 0
2024-08-05T17:21:36.743006764-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:36.743137318-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:36.743247514-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:36.743369783-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:36.743898343-07:00 [INFO] activate is not implemented on Linux, ignoring the call
2024-08-05T17:21:36.74404619-07:00 [INFO] Opening main db
2024-08-05T17:21:36.745528425-07:00 [INFO] perform;
2024-08-05T17:21:36.745699776-07:00 [INFO] read_command;
2024-08-05T17:21:36.745825812-07:00 [INFO] read_command;
2024-08-05T17:21:36.745986222-07:00 [INFO] socket reader;
2024-08-05T17:21:36.750824931-07:00 [INFO] set environment variables from shell:/bin/bash, path:/home/barugon/.cargo/bin:/home/barugon/.local/bin:/home/barugon/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin
2024-08-05T17:21:36.759083414-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:36.759205633-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:36.759313224-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:36.759420725-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:36.759549557-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:36.783640251-07:00 [INFO] Enabling Vulkan Portability
2024-08-05T17:21:36.783763922-07:00 [INFO] Enabling color space support
2024-08-05T17:21:36.800877614-07:00 [INFO] Adapter "AMD Radeon RX 7800 XT (RADV NAVI32)"
2024-08-05T17:21:36.801023858-07:00 [INFO] Ray tracing is supported
2024-08-05T17:21:37.103618207-07:00 [INFO] Using surface present mode MAILBOX
2024-08-05T17:21:37.104206067-07:00 [WARN] Unable to forbid exclusive full screen
2024-08-05T17:21:37.105270059-07:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-05T17:21:37.105405843-07:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-08-05T17:21:37.10554272-07:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2024-08-05T17:21:37.111414343-07:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "AMD Radeon RX 7800 XT (RADV NAVI32)", driver_name: "radv", driver_info: "Mesa 24.1.5" }
2024-08-05T17:21:37.112828681-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:37.113012776-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:37.114221389-07:00 [INFO] Opening main db
2024-08-05T17:21:37.196370644-07:00 [INFO] Initializing default prettier with plugins {}
2024-08-05T17:21:37.196950109-07:00 [ERROR] no worktrees when constructing ProjectLspAdapterDelegate
2024-08-05T17:21:37.203642419-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:37.224483149-07:00 [INFO] Using surface present mode MAILBOX
2024-08-05T17:21:37.225289678-07:00 [WARN] Unable to forbid exclusive full screen
2024-08-05T17:21:37.231709298-07:00 [INFO] Initializing default prettier with plugins {}
2024-08-05T17:21:37.236673552-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:37.499232464-07:00 [INFO] Initializing default prettier with plugins {}
2024-08-05T17:21:37.500729517-07:00 [WARN] request completed with error: failed to connect to the server
2024-08-05T17:21:40.568874926-07:00 [INFO] Using surface present mode MAILBOX
2024-08-05T17:21:40.56963535-07:00 [WARN] Unable to forbid exclusive full screen
2024-08-05T17:21:42.419094134-07:00 [INFO] Using surface present mode MAILBOX
2024-08-05T17:21:42.419821787-07:00 [WARN] Unable to forbid exclusive full screen
</pre></details>

**Comment by m4saurabh** [comment]
What package manager did you use to install Zed

**Comment by Barugon** [comment]
> @Barugon What package manager did you use to install Zed

I downloaded the tar.gz from here.

**Comment by tetafro** [comment]
Have the same issue - tiny cursor in Zed. Probably related - I have system interface scale set to 200% (Settings - Displays - Scale).

The OS is also Ubuntu 22.04.

**Comment by andmart** [comment]
I have a problem similar. In my case, the cursor is larger! Ubuntu 22.04

And the answers for the commands above my theme: (<<'Yaru'>>,), (<<24>>,)

**Comment by PedroSouza157998** [comment]
I have the same problem. I use Ubuntu 22.04 and the cursor is larger.

**Comment by edwin-epr** [comment]
i have the same problem, i have ubuntu 22.04, the cursor is larger.

**Comment by jeanctech** [comment]
i have the same problem, i have Debian 12.7 - Gnome 43.9 , the cursor is larger.

**Comment by jilherme** [comment]
Same problem on Pop_Os 22. Cursor is larger. 100% display scale

**Comment by veritymedia** [comment]
Same problem on Zorin OS 17.1 Core. Cursor is much larger only in Zed. If I move it to the app bar at the bottom, but becomes normal size.

**Comment by AyushDahal10** [comment]
I have the same problem as well. Please resolve this problem ASAP.

**Comment by watsoncj** [comment]
Ubuntu 22, 100% display scale. Large mouse cursor in stable, preview, flatpak and when built from main (69e698c3be).

Mouse cursor is off by several pixels making it difficult to click targets such as the tab close icon and resize handles.

Have not been able to reproduce the large mouse with the gpui/examples.

**Comment by watsoncj** [comment]
This may be a duplicate of #17771

Workaround by enabling fractional scaling. See https://github.com/zed-industries/zed/issues/17771#issuecomment-2366824512

Or reverting the breaking commit:
```
git revert c1193875e8e93746379514e9ce4fa3db5a0b6503
./script/install-linux
```

**Comment by PedroSouza157998** [comment]
following this line I execute that command: `gsettings set org.gnome.mutter experimental-features "['scale-monitor-framebuffer']"`.
It solved the problem to me.

**Comment by Barugon** [comment]
Enabling fractional scaling did not resolve this for me (Fedora 40).

**Comment by ismaelxyz** [comment]
I was looking for information, I'm glad that at least there is a topic about it, after updating Zed, when entering the app I saw that the cursor was bigger than normal but only inside the app

**Comment by snipchu** [comment]
I had this issue on ubuntu 22.04 too! I only needed to execute the command @PedroSouza157998 put down without doing anything else

**Comment by edwin-epr** [comment]
I use the command of @PedroSouza157998  but not work for me :(. I've got Ubuntu 22.04 in a laptop.

**Comment by TalhaAgteks** [comment]
@ismaelxyz i try run command in terminal but not work for me :|

**Comment by ismaelxyz** [comment]
@TalhaAgteks it is worrying to know that this command does not work for everyone, even if it is the same OS

**Comment by TalhaAgteks** [comment]
@ismaelxyz @PedroSouza157998 i run code terminal again and restart pc fix this issue thx

**Comment by JosephTLyons** [comment]
Can people in this thread check the latest zed preview and see if the mouse pointer issue is fixed? We've done some mouse pointer size fixes, just not sure what all cases it covers.

https://zed.dev/releases/preview/latest

**Comment by watsoncj** [comment]
Tested 54683ff2b9dfcb0c8085aeac021439cafb69a1bf on Ubuntu 22.04. Gnome 42.9.

Behavior has been corrected for the first case.

Scale | Fractional Scaling | Pointer Size Pass/Fail | Hotspot Pass/Fail | Notes
--|--|--|--|--
100% | Off | 🟢 Pass | 🟢 Pass | Cursor 100%, Precise Hotspot
100% | On | 🟢 Pass | 🟢 Pass | Cursor 100%, Precise Hotspot
200% | Off | 🔴 Fail | 🔴 Fail | Cursor 100% (should be 200%), Hotspot too far left and up (~10px / ~10px)
200% | On | 🟢 Pass | 🟢 Pass | Cursor 200%, Precise Hotspot

Hotspot tested on window and split resize handles.

**Comment by ali-42** [comment]
I just experimented a cursor size problem. Mouse cursor looks a lot bigger (maybe twice the size) on Zed than on the rest of the desktop

The problem is present on wayland but not on Xorg.
I'm using debian 12 on the GNOME desktop environment. Scale is 100%
Zed version : 0.156.2.

**Comment by apricotbucket28** [comment]
> Tested [54683ff](https://github.com/zed-industries/zed/commit/54683ff2b9dfcb0c8085aeac021439cafb69a1bf) on Ubuntu 22.04. Gnome 42.9.
> 
> Behavior has been corrected for the first case.
> Scale 	Fractional Scaling 	Pointer Size Pass/Fail 	Hotspot Pass/Fail 	Notes
> 100% 	Off 	🟢 Pass 	🟢 Pass 	Cursor size 100%, precise hotspot
> 100% 	On 	🟢 Pass 	🟢 Pass 	Cursor size 100%, precise hotspot
> 200% 	Off 	🔴 Fail 	🔴 Fail 	Cursor size 100% (should be 200%), hotspot too far left and up (~10px, ~10px)
> 200% 	On 	🟢 Pass 	🟢 Pass 	Cursor size 200%, precise hotspot
> 
> Hotspot tested on window and split resize handles. Gnome session restarted between each test (after applying display settings).

To fix this issue, the cursor's scale should take into account the surface scale (which it doesn't right now), see: https://github.com/zed-industries/zed/pull/17496#issuecomment-2364505645. It's not a big change.

For more information on cursors on Wayland: https://blogs.kde.org/2024/10/09/cursor-size-problems-in-wayland-explained/.
